### PR TITLE
Implement livox avia lidar rays

### PR DESCRIPTION
**Pull request by KristjanDeepX**
### Description

Add livox avia rays to Ogre2GpuRays. We write into the cubemap sampling buffer every frame, instead of just during creation. Each iteration sets new rays according to the avia algorithm. We also pass the ray directions in the image buffer, since the directions aren't static anymore and thus cannot deduce them upstream.

### Motivation and Context

Simulating a real lidar pattern gives us better fidelity to the real world.

### How Has This Been Tested?

Unit tests.

### Checklist

- [x] Have you filled the sections above?
- [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same GitHub Issues/ClickUp Task(s)?
- [x] Have you documented the changes in GitHub Issues and/or ClickUp Task(s) related to this Pull Request?
- [x] Have you added labels, where appropriate, to this Pull Request?

### CODE REVIEW

File: test/integration/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,6 @@
 set(TEST_TYPE "INTEGRATION")
 
 set(tests
-  gpu_rays.cc
```
Comment (nicholasdeepx) [inline test/integration/CMakeLists.txt:4]: Sorry stupid question: why are `gpu_rays.cc` and `lidar_visual.cc` (below) excluded from the build? Is it related to the test problem which you mentioned?

File: src/RenderingIface_TEST.cc [hunk 2]
```diff
@@ -63,7 +63,7 @@ TEST(RenderingIfaceTest, GetEngine)
   EXPECT_EQ(nullptr, sceneFromFirstRenderEngine());
 
   // check get engine
-  for (unsigned int i = 0; i < count; ++i)
```
Comment (nicholasdeepx) [inline src/RenderingIface_TEST.cc:66]: Which engines do `i=0` and `1` here refer to? Why is `0` excluded here?

File: ogre2/src/Ogre2GpuRays.cc [hunk 3]
```diff
@@ -694,56 +711,104 @@ void Ogre2GpuRays::CreateSampleTexture()
...
+
+  this->dataPtr->timer.Start();
+}
+
+/////////////////////////////////////////////////////////
+void Ogre2GpuRays::UpdateSampleTexture()
+{
+  auto getLidarPoints = [](double timepoint, double timespan, size_t pointCount)
+  {
+    constexpr size_t arrayCount = 6;
```
Comment (nicholasdeepx) [inline ogre2/src/Ogre2GpuRays.cc:728]: What does the value `6` here mean?

**Review by nicholasdeepx — APPROVED** [review]
LGTM, but I just have some questions to better understand it (as usual).

File: ogre2/src/Ogre2GpuRays.cc (hunk 3 above)
Comment (KristjanDeepX) [reply ogre2/src/Ogre2GpuRays.cc:728]: It's the laser array count in the avia multi-line laser.
https://terra-1-g.djicdn.com/65c028cd298f4669a7f0e40e50ba1131/Download/Avia/MID70Pro-1.mp4

File: src/RenderingIface_TEST.cc (hunk 2 above)
Comment (KristjanDeepX) [reply src/RenderingIface_TEST.cc:66]: 0 is Ogre1 (which for whatever reason doesn't work), and 1 is Ogre2.

File: test/integration/CMakeLists.txt (hunk 1 above)
Comment (KristjanDeepX) [reply test/integration/CMakeLists.txt:4]: All of the tests assume the old laser pattern, which doesn't apply to us.
We might want to write some unit tests in the future which do apply to us, but we would need to write them from scratch.

Comment (nicholasdeepx) [reply test/integration/CMakeLists.txt:4]: I see. OK, unless you have further plans, let's exclude them for the first release then. We can come to it again for the subsequent release. We may also ask Perception if they have some criteria for the unit tests.

**Pull request merged**